domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -3221,7 +3322,7 @@ public void selectSingle(AsyncResult<SQLConnection> conn, String sql, Tuple para
    * @param chunkSize cursor fetch size
    */
   public Future<Void> selectStream(String sql, Tuple params, int chunkSize, Handler<RowStream<Row>> rowStreamHandler) {
-    return withTrans(trans -> trans.selectStream(sql, params, chunkSize, rowStreamHandler));
+    return withReadTrans(trans -> trans.selectStream(sql, params, chunkSize, rowStreamHandler));
```

Comment:
sql can be a writing SQL: `UPDATE foo SET jsonb=$1 RETURNING jsonb`

Therefore this change cannot be made in this `selectStream` method but requires a new `selectReadStream` method.